DataDog/datadog-agent
# Add assured flag to udp connections

### What does this PR do?

This PR completes https://github.com/DataDog/datadog-agent/pull/5358, adding separate timeouts for udp assured vs non-assured connections and adding tests. We currently use the higher "stream" udp timeout of 120s (typically) to mark a udp connection as expired. This change will allow us to use the lower timeout for non-assured udp connections, which is typically 30s. This in turn should result in a lower memory footprint since we will expire more connections per connections check, since the assured flag is usually not set for most udp connections.

### Motivation

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

1. Run the system-probe
2. Run `socat UDP4-LISTEN:8888,fork EXEC:cat` to run a udp echo server
3. In another console, run `socat - UDP4:localhost:8888` to run a udp client. Type in at least two lines of text and hitting enter after each line. The text for each line should echo back
4. Run `curl -s --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections?client_id=1  | jq '.conns[] | select(.type=="udp" and (.laddr.port==8888 or .raddr.port==8888))'`. You should see two connections
5. Wait a little over 30 seconds. Run (4.) again, you should still see two connections
6. Run (5.) again every 30 seconds or so. After about 2 minutes total, you should see an empty response from the command in (4.)
